### DISCUSSION TITLE
Hide Snippets candidates when inside a function / method parameter

### DISCUSSION THREAD

**Discussion by rodhash** [hrsh7th/nvim-cmp#2015] [General]
`My issue:` I see a weird behavior, while typing in a function parameter I see snippets candidates

<img width="435" alt="image" src="https://github.com/user-attachments/assets/418b2dab-0aac-479d-8098-42aba63352a1">

`So far I tried..`
Tried few different solutions with `priority` and `group_index` in `sources` but no luck so far.

It doesn't seem right to see snippets candidates while we're typing a function parameter.. by chance is this behavior expected?

Something that I noticed:

This only happens if I enter a space in there right after `(` .. without that space char the Snippets don't show up (as expected):
<img width="441" alt="image" src="https://github.com/user-attachments/assets/26c8ee94-472c-40e2-b400-f52b3fff2627">

Still, feels weird to see snippets there. Any suggestion how to avoid this even if we type in a `space` after `(`?

I don't think this could be affecting this behavior but this is my current `setup.sorting` setting:
```lua
      cmp.setup {
        sorting = {
          comparators = {
            compare.order,
            compare.exact,
            compare.score,
            compare.locality,
...
```

**Comment by rodhash** [comment]
Hmmm maybe it was not the optimal order, exact and score seems to be working fine

**Comment by rodhash** [comment]
The order now is fixed and I see parameters before snippets but I still think snippets and probably even buffers words shouldn't appear in here.. is there a specific config to change this behavior? I mean, CMP to bring only parameters when I'm filling up a function params?

<img width="338" alt="image" src="https://github.com/user-attachments/assets/95ceb7fd-7ee9-41dc-85f5-9c4310652895">


I was able to remove the buffers words candidates when the LSP candidates are showing up but this makes buffers words to not show up at all, which is not ideal too

**Reply by Shougo** [reply]
You can see the order.

https://github.com/hrsh7th/nvim-cmp/discussions/1817
